src/functions/warnOnce.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+var bind = require('lodash/function/bind');
+
+var warn = window ?
+  window.console && bind(window.console.warn, console) :
+  bind(console.warn, console); // eslint-disable-line no-console
+
+var warnOnce = (function(w) {
+  var previousMessages = [];
```

Comment:
Should be ok since it only grows on the number of "unknown" params, not by the number of calls to setQueryParameter etc..